frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/cell/BranchOrCommitCellRendererComponent.java

```diff
@@ -160,6 +161,10 @@ public BranchOrCommitCellRendererComponent(
 
       if (hasBranchActionToolTips && branch.isNonRoot()) {
         setToolTipText(getSyncToParentStatusBasedToolTipText(branch.asNonRoot()));
+      } else if (branch.isRoot()) {
+        setToolTipText(getRootToolTipText(branch.asRoot()));
+      } else {
+        setToolTipText(getDefaultNonRootToolTipText(branch.asNonRoot()));
```

Comment:
Hmm actually I know that the original issue (#651) has a point `ensure that some tooltip is being provided in the all other cases`... but now that I checked, `hasBranchActionToolTips` is deliberately set to `false` in `com.virtuslab.gitmachete.frontend.actions.toolbar.DiscoverAction#actionPerformed` so that tooltips are NOT present in the Discover dialog (to avoid cognitive overload on the user).

Reply:
So... what I'd do here is rather skip setting tool tip text if `hasBranchActionToolTips` is false... possibly while you're here, this param should be renamed everywhere to `shouldDisplayActionToolTips` for clarity?

Reply:
Sounds good. Will update PR then.